src/classes/RD2_RecurringDonationsGateway.cls

```diff
@@ -0,0 +1,131 @@
+* @date 2019
+* @group Recurring Donations
+* @description Recurring Donations Gateway to return data
+*
+*/
+public with sharing class RD2_RecurringDonationsGateway {
+
+    /***
+    * @description Retrieves Recurring Donation records for specified Account Ids
+    * @param accountIds Account Ids
+    * @return List<npe03__Recurring_Donation__c>
+    */
+    public List<npe03__Recurring_Donation__c> getRecordsReferencingAccounts(Set<Id> accountIds) {
+        if (accountIds == null) {
+            return new List<npe03__Recurring_Donation__c>();
+        }
+
+        accountIds.remove(null);
+        if (accountIds.isEmpty()) {
+            return new List<npe03__Recurring_Donation__c>();
+        }
+
+        return [
+            SELECT 
+                Name, 
+                npe03__Organization__c,
+                npe03__Contact__c
+            FROM npe03__Recurring_Donation__c 
+            WHERE npe03__Organization__c IN :accountIds
+            AND npe03__Contact__c = null
+        ];
+    }
+    
+    /***
+    * @description Retrieves Recurring Donation records for specified Contact Ids
+    * @param contactIds Contact Ids
+    * @return List<npe03__Recurring_Donation__c>
+    */
+    public List<npe03__Recurring_Donation__c> getRecordsReferencingContacts(Set<Id> contactIds) {
+        if (contactIds == null) {
+            return new List<npe03__Recurring_Donation__c>();
+        }
+
+        contactIds.remove(null);
+        if (contactIds.isEmpty()) {
+            return new List<npe03__Recurring_Donation__c>();
+        }
+        
+        return [
+            SELECT 
+                Name, 
+                npe03__Organization__c,
+                npe03__Contact__c
+            FROM npe03__Recurring_Donation__c 
+            WHERE npe03__Contact__c IN :contactIds
+        ];
+    }
+
+    /***
+    * @description Retrieves Recurring Donation record for the specified Id
+    * @param rdId Id of the recurring donation
+    * @return npe03__Recurring_Donation__c
+    */
+    public npe03__Recurring_Donation__c getRecord(Id rdId) {
+        return getRecords(new Set<Id>{ rdId })[0];
+    }
+
+    /***
+    * @description Retrieves Recurring Donation records for the specified records
+    * @param List<npe03__Recurring_Donation__c> Recurring donations
+    * @return List<npe03__Recurring_Donation__c>
+    */
+    public List<npe03__Recurring_Donation__c> getRecords(List<npe03__Recurring_Donation__c> rds) {
+        if (rds == null || rds.isEmpty()) {
+            return new List<npe03__Recurring_Donation__c>();
+        }
+
+        return getRecords(new Map<Id, npe03__Recurring_Donation__c>(rds).keySet());
+    }
+
+    /***
+    * @description Retrieves Recurring Donation records for the specified Ids
+    * @param rdIds Ids of the recurring donations
+    * @return List<npe03__Recurring_Donation__c>
+    */
+    public List<npe03__Recurring_Donation__c> getRecords(Set<Id> rdIds) {
+        return [
+            SELECT 
```

Comment:
Are there other fields that should be in this query, or better yet should it query *all* fields on the object? It probably depends on usage, but I get the impression this is a general selector purpose method to retrieve RDs.

Reply:
The class has been removed and replaced with test `TEST_SObjectGateway` in order to retrieve test records. Different fields based on the Enhanced RD enablement will be done in the child branch PR.

Reply:
Yes, that can work.